lib/rubocop/cop/rspec/described_class.rb

```diff
@@ -38,27 +52,42 @@ def on_block(node)
           describe, described_class, body = described_constant(node)
           return unless top_level_describe?(describe)
 
-          find_constant_usage(body, described_class) do |match|
-            add_offense(match, :expression, format(MSG, match.const_name))
+          @described_class = described_class
```

Comment:
Yeah I actually have sometimes bitten the bullet before and just called the node traversal calls twice. That is a good reason though. Maybe a comment above the ivar assign then would be helpful?

Also, although we have a cop that discourages this, what happens if we have something like this?

``ruby
RSpec.describe Foo do
  it 'blah' do
    described_class
  end
end

RSpec.describe Bar do
  it 'blah' do
    described_class
  end
end
```